example/rlhf/mlflow/supervised_finetuning_with_mlflow_demo.py

```diff
@@ -0,0 +1,55 @@
+"""Demo for the supervised fine tuning.
+
+python -m example.rlhf.supervised_finetuning_demo
+"""
+
+from pykoi.chat import QuestionAnswerDatabase
+from pykoi.rlhf import RLHFConfig
+from pykoi.rlhf import SupervisedFinetuning
+import mlflow
```

Comment:
To make sure other people from the community can benefit from your amazing work.

- Make sure you add MLflow as a dependency for the pykoi package after [here](https://github.com/CambioML/pykoi/blob/main/pyproject.toml#L44) following the same pattern. 
- Then, add it to the subpackage after [here](https://github.com/CambioML/pykoi/blob/main/pyproject.toml#L72)